src/rules/typescript.ts

```diff
@@ -8,6 +8,8 @@ import memberOrderingList from './rule-helpers/member-ordering-list';
 export default {
   // Enforces function overloads to be consecutive.
   'adjacent-overload-signatures': true,
+  // Bans specific types from being used. Does not ban the corresponding runtime objects from being used.
+  'ban-types': false,
```

Comment:
This may be useful, possibly. 
Any suggestions on specific keywords to ban as type names?
[See here for test case examples](https://github.com/palantir/tslint/tree/master/test/rules/ban-types)